File: src/data_structure_graph/GraphArray.java
package data_structure_graph;

import java.util.Queue;

/**
 * Created by parkdongjo on 2017. 9. 13..
 */
public class GraphArray {

    private static final int WORK_FAIL = -1;
    private static final int WORK_SUCCESS = -1;
    private static final int IN_VERTEX = 1;

    private Integer[][] matrix;
    private Integer[] array;
    private int cursor = 0;
    private int size;

    public GraphArray(int size) {
        this.size = size;
        this.matrix = new Integer[size][size];
        this.array = new Integer[size];
    }

    public void graphInit() {
        for(int i=0; i<matrix.length; i++) {
            for(int j=0; j<matrix.length; j++) {
                matrix[i][j] = null;
            }
        }
        for(int i=0; i<array.length; i++) {
            array[i] = null;
        }
    }

    public int insertVertex(int v) {
        int result = WORK_FAIL;

        if(cursor < this.size) {
            array[++cursor] = v;
            result = WORK_SUCCESS;
        }
        return result;
    }

    public int insertEdge(int v1, int v2) {
        int result = WORK_FAIL;

        if(isEmpty(v1,v2)) {
            matrix[v1][v2] = IN_VERTEX;
            matrix[v2][v1] = IN_VERTEX;
            result = WORK_SUCCESS;
        }

        return result;
    }

    public int deleteVertex(int v) {
        int result = WORK_FAIL;

        for(int i=0; i<array.length; i++) {
            if(array[i] == v) {
                array[i] = null;
                result = WORK_SUCCESS;
            }
        }

        return result;
    }

    public int deleteEdge(int v1, int v2) {
        int result = WORK_FAIL;

        if(matrix[v1][v2] != null) {
            matrix[v1][v2] = null;
            result = WORK_SUCCESS;
        }

        return result;
    }

    public boolean isEmpty(int v1,int v2) {
        return matrix[v1][v2] == null ? true : false;
    }

    public int[] Adjacent(int v) {
        int[] result = null;
        int idx = 0;

        for(int i=0; i<size; i++) {
            if(matrix[v][i] == IN_VERTEX) {
                result[idx++] = i;
            }
        }

        return result;
    }
}

File: src/data_structure_graph/GraphArray.kt
package data_structure_graph

fun main(args: Array<String>) {
    var graph = GraphArray(6)

    graph.insertEdge(1,2)
    graph.insertEdge(1,3)
    graph.insertEdge(2,3)
    graph.insertEdge(2,4)
    graph.insertEdge(3,4)
    graph.insertEdge(3,5)
    graph.insertEdge(4,5)
    graph.insertEdge(4,6)

    graph.print()
}

class GraphArray {
    private var _matrix: Array<Array<Int>>
    private var _array: IntArray

    private var _cursor = 0
    var cursor: Int
        get() = this._cursor
        set(i: Int) { this._cursor = i}

    constructor(size: Int) {
        this._matrix = Array<Array<Int>>(size, {Array<Int>(size, {i->0})})
        this._array = IntArray(size)
    }

    fun getVertex(i: Int): Array<Int> {
        return this._matrix[i]
    }

    fun insertVertex(x: Int): Unit {
        this._array[this._cursor++] = x
    }

    fun insertEdge(x: Int, y: Int): Boolean {
        var success: Boolean = true

        if (this._array.filter{it==x}.isEmpty()) {
            this.insertVertex(x)
        }
        if (this._array.filter{it==y}.isEmpty()) {
            this.insertVertex(y)
        }

        this._matrix[x-1][y-1] = 1
        this._matrix[y-1][x-1] = 1
        return success

    }

    fun print(): Unit {
        for (arr in this._matrix) {
            println()
            for (e in arr) {
                print("$e ,")
            }
        }
    }

}